### PR TITLE
Add machine resource to related objects

### PR DESCRIPTION
**Pull request by michaelgugino**
This commit ensures all machines and machinesets
are included in related-resources.

Implements: https://jira.coreos.com/browse/CLOUD-563

### CODE REVIEW

File: pkg/operator/status.go [hunk 1]
```diff
@@ -185,6 +185,12 @@ func (optr *Operator) getOrCreateClusterOperator() (*osconfigv1.ClusterOperator,
 			Resource: "namespaces",
 			Name:     optr.namespace,
 		},
+		{
```
Comment (enxebre) [inline pkg/operator/status.go]: sweet!
can we ad a PR desc?
can you confirm "" brings all the existing objects?
can we please also add machineSets?

Comment (michaelgugino) [reply pkg/operator/status.go]: @enxebre https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_machine-api-operator/391/pull-ci-openshift-machine-api-operator-master-e2e-aws/1448/artifacts/e2e-aws/must-gather/namespaces/openshift-machine-api/machine.openshift.io/

You can see we're gathering machines and machinesets here.

File: pkg/operator/status.go [hunk 2]
```diff
@@ -185,6 +185,18 @@ func (optr *Operator) getOrCreateClusterOperator() (*osconfigv1.ClusterOperator,
...
+			Resource:  "machines",
+			Name:      "",
+			Namespace: optr.namespace,
+		},
+		{
+			Group:     "machine.openshift.io",
+			Resource:  "machinesets",
+			Name:      "",
+			Namespace: optr.namespace,
+		},
```
Comment (russellb) [inline pkg/operator/status.go]: If you end up having to rebase for some reason, could you add the `baremetalhosts` resource from group `metal3.io`, as well?

Comment (ingvagabund) [reply pkg/operator/status.go]: @russellb we don't need to rebase (pull new code under vendor) at all, right? Just open another PR and extend the list with:
```
{
	Group:     "metal3.io",
	Resource:  "baremetalhosts",
	Name:      "",
	Namespace: optr.namespace,
},
```

**Pull request merged**